content/ja/handbook/engineering/architecture/design-documents/ai_catalog/index.md

```
@@ -0,0 +1,804 @@
+### 外部エージェントの実行
+
+外部エージェント（third-party フロー）は Duo Workflow Service を **使用しません**。CI ワークロードとして直接実行されます:
+
+1. `Ai::FlowTriggers::RunService` がトリガーイベントを受信
+2. フロー定義（Docker イメージ、コマンド）が `ItemVersion#definition` から読み取られる
+3. `Ci::Workloads::RunWorkloadService` が CI ジョブを作成
+4. コンテキストが `AI_FLOW_*` 環境変数で渡される
+
+## エージェントアイデンティティ
+
+フローと外部エージェントが [Flow Triggers](#flow-) を通じてランナー上で実行される時、エージェントの権限は [複合アイデンティティ](https://docs.gitlab.com/user/duo_agent_platform/composite_identity/) を通じて付与されます。
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)